--- a/code
+++ b/code
@@ -60,6 +60,20 @@ for i in "${SDK[@]}"; do
   fi
 done

+# https://github.com/flathub/com.visualstudio.code/pull/427
+export PATH="${EXTRA_PATH}:${PATH}"
+
+# https://github.com/flathub/com.visualstudio.code/pull/428
+if [ ! -e /etc/static/shells ] && [ -e /var/run/host/etc/static/shells ]; then
+  mkdir -p /etc/static/
+  ln -s /var/run/host/etc/static/shells /etc/static/shells
+fi
+
+# https://github.com/flathub/com.visualstudio.code/pull/476
+if [ -f "${XDG_CONFIG_HOME}/code-flags.conf" ]; then
+  mapfile -t USER_FLAGS <<< "$(grep -v '^#' "${XDG_CONFIG_HOME}/code-flags.conf")"
+fi
+
 if [ ! -e /etc/shells ] && [ -e /var/run/host/etc/shells ]; then
   ln -s /var/run/host/etc/shells /etc/shells
 fi
@@ -67,4 +81,4 @@ fi
 exec env ELECTRON_RUN_AS_NODE=1 PATH="${PATH}:${XDG_DATA_HOME}/node_modules/bin" \
   /app/bin/zypak-wrapper.sh /app/extra/vscode/code /app/extra/vscode/resources/app/out/cli.js \
   --ms-enable-electron-run-as-node --extensions-dir=${XDG_DATA_HOME}/vscode/extensions \
-  "$@" ${WARNING_FILE}
+  "${USER_FLAGS[@]}" "$@" ${WARNING_FILE}
